scitt_emulator/rkvst.py

```diff
@@ -0,0 +1,141 @@
+        print(f'>>> Got Operation ID {operation_id}')
+
+    def create_receipt_contents(self, countersign_tbi: bytes, entry_id: str):
+        # TODO: the emulator passes in the full countersigned thing but our interface
+        # takes just the claim. Need to work out if this is the right place to strip
+        # off the furniture or whether our receipt interface should change.
+        # For the hackathon it's bodged to just pass the claim straight back in, and
+        # we can work out how to do it properly during the event
+
+        # TODO: There are a number of inconsistencies between RKVST Cose and client Cose encoding of these things:
+        # * RKVST: CBORTag(18,                [b'\xa3\x01&\x03papplication/json\x19\x01\x87mapp.rkvst.com', {},                                                                       b'{"identity":"assets/71ae3f6e-6228-4e1b-9acb-7883d1b006ad/events/5b2bc2fc-929f-4858-964f-ff1d1ada6e5c"}',  b'\xd6\xfe\xb54<\xfd\xea\x11\xe0\xd5@)\xab\x9c_\x18\xa3n=\x85d\xf8\x01\xd1d\xd3z$c<\xa4\xc2W\xf0\xb0\x11\x8e\xa1o\xee[h\x80\xf7hMw\x8f|\xd0\\\x84x\xd6\xd5\x9f\xf1z\x1e\xe0o4\xa7L'])
+        # * TBI:   ['CounterSignatureV2',      b'\xa3\x01&\x03papplication/json\x19\x01\x87mapp.rkvst.com',     b'\xa3jservice_ideRKVSThtree_algeRKVSTiissued_at\x1ad\x1e\x86"', b'', b'{"identity":"assets/71ae3f6e-6228-4e1b-9acb-7883d1b006ad/events/5b2bc2fc-929f-4858-964f-ff1d1ada6e5c"}', [b'\xd6\xfe\xb54<\xfd\xea\x11\xe0\xd5@)\xab\x9c_\x18\xa3n=\x85d\xf8\x01\xd1d\xd3z$c<\xa4\xc2W\xf0\xb0\x11\x8e\xa1o\xee[h\x80\xf7hMw\x8f|\xd0\\\x84x\xd6\xd5\x9f\xf1z\x1e\xe0o4\xa7L']]
+        #
+        # The important differences are: 
+        #  * presence of a CBORTag outer envelope;
+        #  * more entries in the countersign;
+        #  * sigs are treated as a list in tbi, but a singleton in RKVST
+        #
+        # We beleive the RKVST implementation to be more correct so have left this here for compatibility until the emulator is fixed.
+        [cstr, source, csissuer, phdr, payload, sig]=cbor2.loads(countersign_tbi)
+        cose_contents=cbor2.loads(countersign_tbi)
+        encoded_claim_str=countersign_tbi
+        rkvst_claim_contents = [
+            source,
+            {},
+            payload,
+            sig[0],
+        ]
+        # TODO: b'\xd2' is a hack to get the CBORTag(18) onto the front
+        rkvst_claim_cbor = b'\xd2' + cbor2.dumps(rkvst_claim_contents)
+        encoded_claim = base64.b64encode(rkvst_claim_cbor)
+        encoded_claim_str = str(encoded_claim)
+        encoded_claim_str = encoded_claim_str[2:-1]
+        
+        rkvst_receipt = self.rkvst_connection.post(
+            f"{self.rkvst_network_fqdn}/archivist/v1/notary/receipts",
+            {"claim": encoded_claim_str},
+        )
+        #rkvst_receipt = rkvst_mocks.mock_receipt
+
+        # TODO: This is just neat debug. Get the JSON form of the receipt
+        receipt_file_path = f'{entry_id}.receipt.json'
```

Comment:
we should really document a json form for them, we do something similar

Reply:
Happy to contribute this. Where do you think we should put it? I'd imagine just a subsection the API portion of the architecture doc.